UserController - real implementation (JDP200101-25)

deseff opened the pull request.

deseff committed:
UserController - real implementation (JDP200101-25)

mryndak commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -2,26 +2,55 @@
-    @RequestMapping(method = RequestMethod.POST, value = "createUser")
-    public void createUser(UserDto userDto) {
+    @RequestMapping(method = RequestMethod.GET, value = "getUser")
+    public UserDto getUser(@RequestParam long id) throws UserNotFoundException {
```
może nie być tutaj throws

deseff replied:
A czy to nie jest tak, że skoro w UserRepository jest:
`@Override
    Optional<User> findById(Long id);`
w UserController metoda getUser powinna wyrzucać wyjątek?
Metoda createUser nie wyrzuca błędu.

mryndak commented on src/main/java/com/kodilla/ecommercee/service/UserDbService.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    public void blockUser(final User user) {
+        if (userRepository.existsById(user.getId())) {
```
Generalnie może tak zostać, ale jakbyś miał czas i możliwości to obejrzyj sobie:
https://www.youtube.com/watch?v=tKfDdtjYYi8
i zastanów się jakby to można było "inaczej" zapisać.
Na kursie nie poruszamy zagadnień związanych z transakcjami, ale warto mieć świadomość, że coś takiego jest i to coś w jakiś sposób wpływa na nasz kod

deseff committed:
UserController - real implementation (JDP200101-25)
- removing "throws UserNotFoundException" from class UserController, method getUser
